### PR TITLE
Add AppGateway Builder

### PR DESCRIPTION
**Pull request by wzamlynny**
This PR starts adding the App Gateway Builder and well as some tests to ensure syntax works.
Still additional work required to complete adding all properties, as well as more extensive tests.

Added properties: 

- Sku
- IP config
- Frontend IP config
- Backend Address Pools
- Backend HTTP Settings

Remaining required properties:

- SSL Certificates
- Probes
- Http Listeners
- Request Routing Rules

Remaining desired properties:

- Zones
- Autoscale
- Web application firewall configuration
- Diagnostics settings
- Project

### CODE REVIEW

File: src/Farmer/Arm/ApplicationGateway.fs [hunk 1]
```diff
@@ -57,7 +57,6 @@ type ApplicationGateway =
            Path: string option
            Port: uint16
            Protocol: Protocol
-           CookieBasedAffinity: FeatureFlag
```
Comment (et1975) [inline src/Farmer/Arm/ApplicationGateway.fs:60]: We don't like cookies?

File: src/Farmer/Builders/Builders.ApplicationGateway.fs [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+            :: backendPools
+            @ (frontendPublicIps |> Seq.cast<IArmResource> |> List.ofSeq)
+            
+
+type AppGatewayBuilder() =
+    member _.Yield _ : AppGatewayConfig = {
+        Name = ResourceName.Empty
+        Sku = {
+            Name = Sku.Standard_v2
+            Capacity = 1 // TODO - what value?
```
Comment (et1975) [inline src/Farmer/Builders/Builders.ApplicationGateway.fs]: ... making it optional.

File: src/Tests/AppGateway.fs [hunk 3]
```diff
@@ -1,30 +1,93 @@
...
+                            "my-auth-cert"
+                        ]
+                        connection_draining (connectionDraining {
+                            drain_timeout 500<Seconds>
+                            enabled true
+                        })
+                        cookie_based_affinity FeatureFlag.Enabled
+                        host_name "my-host-name"
+                        path "my-path"
+                        port 80
```
Comment (et1975) [inline src/Tests/AppGateway.fs]: This maybe an existing pattern, but I'd have preferred if we surfaced `uint16` to the DSL - as it is there would be no compiler error if someone specified `-1` here.

File: src/Farmer/Arm/ApplicationGateway.fs (hunk 1 above)
Comment (wzamlynny) [reply src/Farmer/Arm/ApplicationGateway.fs:60]: It's in the list twice, first instance is at line 55

Comment (et1975) [reply src/Farmer/Arm/ApplicationGateway.fs:60]: Oh, that's so weird that it compiles!

**Pull request merged**